flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java

```diff
@@ -235,7 +235,7 @@ public DataSource createDataSource(Context context) {
             configFactory.chunkKeyColumn(chunkKeyColumnMap);
         }
 
-        return new MySqlDataSource(configFactory);
+        return new MySqlDataSource(configFactory, context.getPipelineConfiguration());
```

Comment:
It's a little weird to pass `PipelineConfiguration` to source since `PipelineConfiguration` should store some pipeline-level global configs, while `rate.limit` here is source-specific.

What about putting `rate.limit` into `source:` block?

Reply:
In my conception, all data sources should support speed limiting. Therefore, this parameter is added globally rather than only for MySQL. (Although currently only MySQL has been implemented.) If it is added in `source:`, then if this function needs to be written for other data sources in the future, will relevant code need to be added again? Wouldn't this be a bit too poor in reusability?

Reply:
I was thinking about putting it into `source` block, but parse it as if it is a pipeline-level config, just like your current implementation.

Your point is very convincing, surely we shouldn't implement rate limit in each source 
connector. Putting it in `pipeline` scope is also reasonable.

Reply:
Good idea，and when I was writing the document, I found that most of the configuration names in the `pipeline ` are like：
- `local-time-zone`
- `user-defined-functio`
use `-` rather than `.`   so，call  `source-rate-limit` maybe is better?

Reply:
Good idea!

Reply:
done